api/query/cache/index/index.go

```diff
@@ -330,6 +334,34 @@ func NewReportLoader() ReportLoader {
 	return HTTPReportLoader{}
 }
 
+func (i *shardedWPTIndex) syncGetRun(id RunID) (shared.TestRun, error) {
+	i.m.RLock()
+	defer i.m.RUnlock()
+
+	run, loaded := i.runs[id]
+	if !loaded {
+		return shared.TestRun{}, errNilRun
+	}
+
+	return run, nil
+}
+
+func (i *shardedWPTIndex) syncGetRuns(ids []RunID) ([]shared.TestRun, error) {
+	i.m.RLock()
+	defer i.m.RUnlock()
+
+	runs := make([]shared.TestRun, len(ids))
+	for j := range ids {
+		run, ok := i.runs[ids[j]]
+		if !ok {
+			return nil, fmt.Errorf("Unknown run ID: %v", ids[j])
```

Comment:
It's a bit counter-intuitive that `Run` and `Runs` will return different types of errors when a run is not loaded. WDYT?